File: legacy/backend/app/routers/journal.py
"""Public read-only journal endpoints — phases, ADRs, glossary."""
from __future__ import annotations

import logging
from typing import Any

from fastapi import APIRouter, Depends, Request
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.db import get_db
from app.limiter import limiter
from app.models import GlossaryTerm, JournalAdr, JournalPhase

router = APIRouter(prefix="/journal", tags=["journal"])
logger = logging.getLogger(__name__)


@router.get("/phases", response_model=list[dict[str, Any]])
@limiter.limit("60/minute")
async def list_phases(
    request: Request,
    db: AsyncSession = Depends(get_db),
) -> list[dict[str, Any]]:
    result = await db.execute(select(JournalPhase).order_by(JournalPhase.phase_number))
    rows = result.scalars().all()
    return [
        {
            "phase_number": r.phase_number,
            "title": r.title,
            "started_at": r.started_at,
            "completed_at": r.completed_at,
            "summary_md": r.summary_md,
            "markdown_file_path": r.markdown_file_path,
            "indexed_at": r.indexed_at,
        }
        for r in rows
    ]


@router.get("/adrs", response_model=list[dict[str, Any]])
@limiter.limit("60/minute")
async def list_adrs(
    request: Request,
    db: AsyncSession = Depends(get_db),
) -> list[dict[str, Any]]:
    result = await db.execute(select(JournalAdr).order_by(JournalAdr.adr_number))
    rows = result.scalars().all()
    return [
        {
            "adr_number": r.adr_number,
            "title": r.title,
            "status": r.status,
            "decided_at": r.decided_at,
            "summary_md": r.summary_md,
            "markdown_file_path": r.markdown_file_path,
            "superseded_by_adr_number": r.superseded_by_adr_number,
            "indexed_at": r.indexed_at,
        }
        for r in rows
    ]


@router.get("/glossary", response_model=list[dict[str, Any]])
@limiter.limit("60/minute")
async def list_glossary(
    request: Request,
    db: AsyncSession = Depends(get_db),
) -> list[dict[str, Any]]:
    result = await db.execute(select(GlossaryTerm).order_by(GlossaryTerm.term))
    rows = result.scalars().all()
    return [
        {
            "term": r.term,
            "category": r.category,
            "plain_definition": r.plain_definition,
            "why_it_matters_in_project": r.why_it_matters_in_project,
            "markdown_file_path": r.markdown_file_path,
            "indexed_at": r.indexed_at,
        }
        for r in rows
    ]

File: legacy/frontend/src/api/journal.ts
import axios from "axios";

const api = axios.create({ baseURL: "/api" });

export interface JournalPhase {
  phase_number: number;
  title: string;
  started_at: string | null;
  completed_at: string | null;
  summary_md: string | null;
  markdown_file_path: string | null;
  indexed_at: string;
}

export interface JournalAdr {
  adr_number: number;
  title: string;
  status: string;
  decided_at: string | null;
  summary_md: string | null;
  markdown_file_path: string | null;
  superseded_by_adr_number: number | null;
  indexed_at: string;
}

export interface GlossaryTerm {
  term: string;
  category: string | null;
  plain_definition: string | null;
  why_it_matters_in_project: string | null;
  markdown_file_path: string | null;
  indexed_at: string;
}

export async function fetchPhases(): Promise<JournalPhase[]> {
  const { data } = await api.get<JournalPhase[]>("/journal/phases");
  return data;
}

export async function fetchAdrs(): Promise<JournalAdr[]> {
  const { data } = await api.get<JournalAdr[]>("/journal/adrs");
  return data;
}

export async function fetchGlossary(): Promise<GlossaryTerm[]> {
  const { data } = await api.get<GlossaryTerm[]>("/journal/glossary");
  return data;
}
